src/codeview.js

```diff
@@ -223,6 +227,9 @@ var Codeview = GObject.registerClass({
     }
 
     set text(value) {
+        if (this._buffer && this._buffer.text === value)
+            return;
```

Comment:
Maybe you need the `GObject.ParamFlags.EXPLICIT_NOTIFY` flag for this to return without notifying?